src/bin/common/solving_loop.ml

```diff
@@ -373,7 +386,8 @@ let main () =
         in
         let rev_cnf = D_cnf.make (State.get State.logic_file st).loc l td in
         let cnf = List.rev rev_cnf in
-        let () = solve all_context (cnf, name) in
+        let solver_ctx = State.get solver_ctx_key st in
```

Comment:
This is somewhat of a nit and I am fine with leaving this as is but for future reference: it is good software engineering to keep transient variable definitions as close to their use as possible. There are two related reasons:

 - It makes the intention (here: to update the solver context in the state) clearer. With this change, the reader needs to figure out whether the calls to `Solver.set` below may be undoing another `State.set` done by a function call or potential side effects between line 389 and line 405 (it doesn't, but the question still needs to be asked). Previously it was clear that no such things was possible, because the call to `State.set` was immediately after the call to `State.get`.
 - If later we *do* need to perform another `State.set` in the middle, having the definition and use of `solver_ctx` far apart increases the risk that we do introduce a bug by unintentionally having the new change reverted.

Reply:
I agree :)